File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int zeroCount = 0;
        int left = 0;
        int right = 0;
        int longest = 0;
        
        while (right < nums.length) {
            if (nums[right] == 0) {
                zeroCount++;
            }
            while (zeroCount == 2) {
                if (nums[left] == 0) {
                    zeroCount--;
                }
                left++;
            }
            longest = Math.max(longest, right - left + 1);
            right++;
        }
        
        return longest;
    }
}

File: 0487-max-consecutive-ones-ii/0487-max-consecutive-ones-ii.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        zeroCount = 0
        left = right = longest = 0
        while right < len(nums):
            if nums[right] == 0:
                zeroCount += 1
            while zeroCount >= 2:
                if nums[left] == 0:
                    zeroCount -= 1
                left += 1
            longest = max(longest, right - left + 1)
            right += 1
        return longest